homeassistant/components/permobil/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The MyPermobil integration."""
+from __future__ import annotations
+
+import logging
+
+from mypermobil import MyPermobil, MyPermobilClientException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_CODE,
+    CONF_EMAIL,
+    CONF_REGION,
+    CONF_TOKEN,
+    CONF_TTL,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+
+from .const import APPLICATION, DOMAIN
+from .coordinator import MyPermobilCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up MyPermobil from a config entry."""
+
+    # create the API object from the config and save it in hass
+    session = hass.helpers.aiohttp_client.async_get_clientsession()
+    p_api = MyPermobil(
+        application=APPLICATION,
+        session=session,
+        email=entry.data.get(CONF_EMAIL),
+        region=entry.data.get(CONF_REGION),
+        code=entry.data.get(CONF_CODE),
+        token=entry.data.get(CONF_TOKEN),
+        expiration_date=entry.data.get(CONF_TTL),
+    )
+    try:
+        p_api.self_authenticate()
+    except MyPermobilClientException as err:
+        _LOGGER.error("Permobil: %s", err)
```

Comment:
The logger will add which integration it is, so remove the "Permobil" part in the beginning.